onboard/nucleo_bridge/LCMHandler.cpp

```diff
@@ -0,0 +1,188 @@
+#include "LCMHandler.h"
+
+void LCMHandler::init() {
+    //Creation of lcm bus
+    lcm_bus = new lcm::LCM();
+    if (!lcm_bus->good())
+    {
+        printf("LCM Bus not created\n");
+        exit(1);
+    } else {
+	printf("LCM Bus created\n");
+    }
+    
+    internal_object = new InternalHandler();
+    
+    //Subscription to lcm channels 
+    lcm_bus->subscribe("/ik_ra_control", &LCMHandler::InternalHandler::ra_closed_loop_cmd, internal_object);
+    lcm_bus->subscribe("/sa_closedloop_cmd", &LCMHandler::InternalHandler::sa_closed_loop_cmd, internal_object);
+
+    lcm_bus->subscribe("/ra_openloop_cmd", &LCMHandler::InternalHandler::ra_open_loop_cmd, internal_object);
+    lcm_bus->subscribe("/sa_openloop_cmd", &LCMHandler::InternalHandler::sa_open_loop_cmd, internal_object);
+
+    lcm_bus->subscribe("/gimbal_openloop_cmd", &LCMHandler::InternalHandler::gimbal_cmd, internal_object);
+    lcm_bus->subscribe("/hand_openloop_cmd", &LCMHandler::InternalHandler::hand_openloop_cmd, internal_object);
+    lcm_bus->subscribe("/foot_openloop_cmd", &LCMHandler::InternalHandler::foot_openloop_cmd, internal_object);
+    /*
+    lcmBus->subscribe("/ra_config_cmd", &LCMHandler::InternalHandler::ra_config_cmd, &internal_object);
+    lcm_bus.subscribe("/sa_config_cmd", &LCMHandler::InternalHandler::sa_config_cmd, internal_object);
+    lcm_bus.subscribe("/ra_zero_trigger", &LCMHandler::InternalHandler::ra_zero_trigger, &internal_object);
+    lcm_bus.subscribe("/sa_zero_trigger", &LCMHandler::InternalHandler::sa_zero_trigger, internal_object);
+    */
+    printf("LCM Bus channels subscribed\n");
+}
+
+//Handles a single incoming lcm message    
+void LCMHandler::handle_incoming()
+{
+    lcm_bus->handle();
+}
+
+//Decide whether outgoing messages need to be sent, and if so, send them
+void LCMHandler::handle_outgoing()
+{
+    //If the last time arm position messages were outputted was over 100 ms ago, get new data from Controllers to be sent
+    std::chrono::duration deadTime = std::chrono::milliseconds(100);

```

Comment:
consider making this `static const`

Reply:
I've tried this before, and the LCM library throws a fit at runtime when the handler object is static.